ci/prepare-documentation.py

```diff
@@ -117,10 +119,17 @@ def add_page(output_root, parent, *, id=None, path=None, title=None):
             continue # not a relative URL
 
         image_rel_path = path.parent / urllib.request.url2pathname(parsed_image_url.path)
+        image_filename = image_rel_path.name
+        image_abs_path = (OMZ_ROOT / image_rel_path).resolve()
 
-        (output_root / image_rel_path.parent).mkdir(parents=True, exist_ok=True)
-        shutil.copyfile(OMZ_ROOT / image_rel_path, output_root / image_rel_path)
+        if image_filename in all_images_paths and all_images_paths[image_filename] != image_abs_path:
+            raise RuntimeError(f'{path}: Image with "{image_filename}" filename already exists. '
+                               f'Rename "{image_rel_path}" to unique name.')
+        else:
+            all_images_paths[image_filename] = image_abs_path
```

Comment:
no need in else statement if condition leads to raising error
```suggestion
        all_images_paths[image_filename] = image_abs_path
```